### PR TITLE
Feature: Notification Handler endpoints

### PR DESCRIPTION
**Pull request by alaca**
## Description
This PR adds endpoints for handling notifications for the new `NotificationPlaceholder` component 
https://github.com/impress-org/form-builder-library/pull/8

## Pre-review Checklist
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: give.php [hunk 1]
```diff
@@ -58,6 +58,7 @@
 use Give\Form\Templates;
 use Give\FormBuilder\ServiceProvider as FormBuilderServiceProvider;
 use Give\FormMigration\ServiceProvider as FormMigrationServiceProvider;
+use Give\NotificationsHandler\ServiceProvider as NotificationHandlerServiceProvider;
```
Comment (jonwaldstein) [inline give.php:61]: Looks like we are giving `NotificationsHandler` it's own top level domain in the src folder.  I'm wondering if this should actually be in the `FormBuilder` domain or if we do want to make this more general like `Notifications` to leave room for other notification type apis.

File: src/NotificationsHandler/Routes/DismissNotification.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use Give\API\RestRoute;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * Dismiss notification route
+ *
+ * @unreleased
+ */
+class DismissNotification implements RestRoute
```
Comment (jonwaldstein) [inline src/NotificationsHandler/Routes/DismissNotification.php:14]: Let's write a test for this 😄 

@pauloiankoski made a `RestApiTestCase` for rest api testing specifically - but you can also do a unit test for `handleRequest` and mock the response like i've done [here](https://github.com/impress-org/givewp/blob/abc1aa7bd0b21d0d301edcfa9046310b64795e04/tests/Feature/Controllers/FormBuilderResourceControllerTest.php#L21)

File: give.php (hunk 1 above)
Comment (alaca) [reply give.php:61]: Good point. For me, this is not entirely the Form Builder thing, but I guess it would make sense to move this to the FB domain.

Comment (alaca) [reply give.php:61]: On another thought, I think having Notifications TLD might be a better option.

**Pull request closed without merging**